cloverage/src/cloverage/coverage.clj

```diff
@@ -177,10 +180,16 @@
           ;; mark the ns as loaded
           (mark-loaded namespace))
         (println "Instrumented namespaces.")
+        (when output
+          (.mkdir (File. output)))
         (let [test-result (when-not (empty? test-nses)
                             (let [test-syms (map symbol test-nses)]
                               (apply require (map symbol test-nses))
-                              (apply test/run-tests (map symbol test-nses))))
+                              (if junit-output
+                                (binding [test/*test-out* (clojure.java.io/writer junit-output)]
```

Comment:
If the JUnit output folder varies from the main output folder, this will break, as it won't make the folder the JUnit output needs. I'd prefer something that always outputs a "junit.xml" into the folder defined by "output" (when the --junit-output option is chosen) rather than having a different output mechanism for JUnit v.s. all other test options.